File: widget/src/main/java/ru/kest/trainswidget/util/DateUtil.java
package ru.kest.trainswidget.util;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.concurrent.TimeUnit;

/**
 * Created by Konstantin on 26.05.2017.
 */

public class DateUtil {

    private static SimpleDateFormat timeFormatter = new SimpleDateFormat("HH:mm");
    private static SimpleDateFormat timeWithSecondsFormatter = new SimpleDateFormat("HH:mm:ss");
    private static SimpleDateFormat dayFormatter = new SimpleDateFormat("yyyy-MM-dd");

    public static String getTime(Date date) {
        return timeFormatter.format(date);
    }

    public static String getTimeWithSeconds(Date date) {
        return timeWithSecondsFormatter.format(date);
    }

    public static String getDay(Date date) {
        return dayFormatter.format(date);
    }

    public static int getTimeDiffInMinutes(Date time) {
        long diffInMillies = System.currentTimeMillis() - time.getTime();
        return (int) TimeUnit.MINUTES.convert(Math.abs(diffInMillies), TimeUnit.MILLISECONDS);
    }
}

File: widget/src/main/java/ru/kest/trainswidget/util/DateUtil.kt
package ru.kest.trainswidget.util

import android.annotation.SuppressLint
import java.text.SimpleDateFormat
import java.util.*
import java.util.concurrent.TimeUnit

/**
 * Date utils
 * Created by Konstantin on 26.05.2017.
 */
@SuppressLint("SimpleDateFormat")
object DateUtil {

    private val timeFormatter = SimpleDateFormat("HH:mm")
    private val dayFormatter = SimpleDateFormat("yyyy-MM-dd")

    fun getTime(date: Date): String {
        return timeFormatter.format(date)
    }

    fun getDay(date: Date): String {
        return dayFormatter.format(date)
    }

    fun getTimeDiffInMinutes(time: Date): Int {
        val diffInMillis = System.currentTimeMillis() - time.time
        return TimeUnit.MINUTES.convert(Math.abs(diffInMillis), TimeUnit.MILLISECONDS).toInt()
    }
}
